### ISSUE TITLE
Provide a theming solution for apps consuming material

### ISSUE THREAD

**Issue by jelbourn** [angular/material2#123]
Needs design doc

**Comment by zoechi** [comment]
angular/angular#2567

**Comment by francisoud** [comment]
@jelbourn about your comment https://github.com/angular/material2/issues/123#issue-136879577 what do you mean by _Needs design doc_
As an end user, I would expect theming to work kind of [angular1](https://material.angularjs.org/latest/Theming/03_configuring_a_theme)  

I would expect  something like this code in a `app/main.ts` since it's a central part of the application:
```
import { bootstrap }      from '@angular/platform-browser-dynamic';
import { mdThemingProvider } from '@angular2-material/core';

import { AppComponent }         from './app.component';
import { appRouterProviders }   from './app.routes';

bootstrap(AppComponent, [
  appRouterProviders,
  mdThemingProvider
]);
```

but in this case I don't know where to put this kind of code

```
mdThemingProvider.theme('default')
    .primaryPalette('pink')
    .accentPalette('orange');
```

so may be another solution it to have 'main.ts' deleguate to another provider like this:
```
import { bootstrap }      from '@angular/platform-browser-dynamic';

import { AppComponent }         from './app.component';
import { appRouterProviders }   from './app.routes';
import { appThemingProvider }   from './app.theme';

bootstrap(AppComponent, [
  appRouterProviders,
  appThemingProvider
]);
```
and a `app.theme.ts`with:
```
import { theme } from '@angular2-material/core';

export const appThemingProvider = [
    theme('default')
        .primaryPalette('pink')
        .accentPalette('orange');
];
```

a alternative would be to reproduce what is done un the [router doc](https://angular.io/docs/ts/latest/guide/router.html)
`app.theme.ts` 
```
import { provideTheme, ThemingConfig } from '@angular2-material/core';

const themes: ThemingConfig = [
  { id: 'default', primaryPalette: 'pink',  accentPalette: 'orange'},
  { ... },
  { ... }
];

export const appRouterProviders = [
  provideTheme(themes);
];

```

**Comment by jelbourn** [comment]
The primary way of theming for material2 will be static. The run-time generation of styles is not suitable for most apps because it is rather expensive. That said, we will possibly also have an optional run-time configuration as well.

**Comment by davidreher** [comment]
maybe it is a stupid question, but can theming be done today at all?

**Comment by jelbourn** [comment]
Not easily until we finish implementing user-facing theming.

**Comment by a5hik** [comment]
Looking forward for this feature!

**Comment by mehran9** [comment]
Same here!

**Comment by ansarizafar** [comment]
1+

**Comment by ReidBraswell** [comment]
+1

**Comment by eserlan** [comment]
+1

Den fre. 12. aug. 2016, 16.06 skrev Reid Braswell <notifications@github.com

**Comment by zakdances** [comment]
It doesn't seem like scss is supported at all right now because components such as sidenav try to import scss with paths like `@import 'default-theme';` which doesn't exist because it's from other components.

**Comment by maciejsobala** [comment]
+1

**Comment by code-tree** [comment]
Thanks for continuing to work on this. I'm not sure if the following is planning on being supported, so I'm mentioning it just in case:

I plan on including a "dark/light theme" setting in my app. I'm hoping I'll be able to export both normal/dark material theme variations, with the dark version only enabled when `body.dark` class is present (set by the UI setting). Will this sort of thing be possible in future?

**Comment by jelbourn** [comment]
Current status: we're trying out a couple of different designs on this now, trying to gauge what works best.

**Comment by HeavenlyHost** [comment]
:+1:

**Comment by a5hik** [comment]
@jelbourn Great! waiting for this to be released!

**Comment by jelbourn** [comment]
Closed by #1239, will be in the next release.
